src/malli/transform.cljc

```diff
@@ -67,11 +67,30 @@
          (catch #?(:clj Exception, :cljs js/Error) _ x))
     x))
 
+(defn parse-float [s]
+  #?(:clj
+     (if (string? s)
+       (try
+         (Float/parseFloat s)
+         (catch NumberFormatException _ nil))
+       (throw (IllegalArgumentException.
+               (str "Expected string, got " (if (nil? s) "nil" (-> s class .getName))))))
+     :cljs
+     (parse-double s)))
+
+(defn -string->float [x]
+  (if (string? x)
+    (or (parse-float x) x)
+    x))
+
```

Comment:
Here I'm providing a custom definition for parsing floats from strings. I could not find a function that used `Float/parseFloat` in this way, so I copied and modified a similar function from Clojure source code.